templates/tfengine/components/cicd/main.tf

```diff
@@ -305,6 +305,16 @@ resource "google_storage_bucket_iam_member" "cloudbuild_state_iam" {
   ]
 }
 
+# IAM permissions to allow Cloud Build SA to access logs bucket.
+resource "google_storage_bucket_iam_member" "cloudbuild_logs_bucket_iam" {
+  bucket = var.logs_bucket
```

Comment:
we should use reference here so logs_bucket is created first before this resource is attempted

Reply:
@xingao267 should this be done for the other IAM permissions in the same file as well then?

Reply:
Depending on whether or not the dependent resource is created in the cicd recipe or not. For example, the state bucket is not created here, but in the `devops` recipe, and the `cicd` recipe takes that as an input, so there is no need to do that for the state bucket. Similarly for the billing account which is not something created in this recipe. However, if the resource is created here, then we should make sure the dependency is sorted out clearly to avoid race conditions.

Reply:
Done.